html/semantics/forms/autofocus/resources/utils.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+function waitForEvent(target, type, options) {
+  return new Promise((resolve, reject) => {
+    target.addEventListener(type, resolve, options);
+  });
+}
+
+function waitForLoad(target) {
+  return waitForEvent(target, 'load');
+}
+
+function timeOut(test, ms) {
+  return new Promise((resolve, reject) => {
+    test.step_timeout(resolve, ms);
+  });
+}
+
+// If an element with autofocus is connected to a document and this function
+// is called, the autofocus result is deterministic after returning from the
+// function.
+// Exception: If the document has script-blocking style sheets, this function
+// doesn't work well.
+async function waitUntilStableAutofocusState(test) {
+  // TODO: Update this for https://github.com/web-platform-tests/wpt/pull/17929
+  await timeOut(test, 100);
```

Comment:
Maybe I'm missing something, but won't this just cause the test to always time out before the rest of the test gets run?

https://github.com/web-platform-tests/wpt/pull/17929/files#diff-c05e2b49bb15cca07f18649490b4e74eR25 has a different implementation.

Reply:
Sorry, I misremembered what `step_timeout` does. Never mind.

Still, which of this PR and #17929 has the implementation you want?

Reply:
> Still, which of this PR and #17929 has the implementation you want?

One in https://github.com/web-platform-tests/wpt/pull/17929 is the final implementation. 
 However, it's not correct until [the specification change for #17929](https://github.com/whatwg/html/pull/4763) is merged.